### PR TITLE
Add a new variable for the wigosid to the bufr2ioda code(s)

### PR DESCRIPTION
**Pull request by PraveenKumar-NOAA**
## Description

This PR adds a new Wigos variable like dateTime to the bufr2ioda code.

_**The following files are new:**_ 

- ~/src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp
- ~/src/bufr/BufrParser/Exports/Variables/WigosidVariable.h
- ~/test/testinput/bufr_ncep_wigos.yaml
- ~/test/testoutput/rtma_ru.t0000z.adpsfc.wigos.nc

_The following files have been changed to add the functionality:_ 

- ~/src/bufr/BufrParser/Exports/Export.cpp
- ~/src/bufr/CMakeLists.txt
- ~/src/bufr/DataObject.h
- ~/test/CMakeLists.txt

_The wigos variable is look like this:_ 
WGOSIDS-WGOSISID-WGOSISNM-WGOSLID, e.g. 0-20000-0-71941

where the naming convention is **stationWIGOSId** (already exist in the ObsSpace.yaml). 

_The ioda-validate.x is also ran over the output NetCDF file and the results are:_ 
**Final results:
errors:      0
warnings:    0**

## Issue(s) addressed

The following issue is addressed: #1402 

## Dependencies

_bufr2ioda_ and **feature/query_new_result** branch of the ioda-converters

## Impact

None

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

**Review by rmclaren — COMMENTED** [review]
It looks like you are mixing code from a mixture of different branches, so you are inadvertently trying to merge long_str branch changes into develop... I think you will need to create a clean branch based on feature/query_new_result or develop and go from there...

File: src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        }
+
+        for (unsigned int idx = 0; idx < map.at(getExportKey(ConfKeys::Wgosids))->size(); idx++)
+        {
+            auto wgosids = map.at(getExportKey(ConfKeys::Wgosids))->getAsInt(idx);
+            auto wgosisid = map.at(getExportKey(ConfKeys::Wgosisid))->getAsInt(idx);
+            auto wgosisnm = map.at(getExportKey(ConfKeys::Wgosisnm))->getAsInt(idx);
+            auto wgoslid = map.at(getExportKey(ConfKeys::Wgoslid))->getAsString(idx);
+            
+	    std::stringstream wgos_all;
```
[the remark this answers is not in the document]
Comment (PraveenKumar-NOAA) [reply src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp:81]: @rmclaren thanks, which one you suggest feature/query_new_result or develop?

File: src/bufr/CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # This software is licensed under the terms of the Apache Licence Version 2.0
 # which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
 
-if ( iodaconv_bufr_query_ENABLED )
+if ( iodaconv_bufr_ENABLED )
```
Comment (nicholasesposito) [inline src/bufr/CMakeLists.txt:6]: Why is this part changed?

File: src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmclaren) [reply src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp:81]: Develop would be fine.

**Comment by YoulongXia-NOAA** [comment]
@PraveenKumar-NOAA, as WIGOS ID is different from dateTime (dateTime is no missing), the wigos ID has missing values. If any part of the four component is missing, it may lead to a string like as ---20000-0-71941, 0----0-71941, etc. Now I know that we can replace WGOSLID with LSTN (station local name) if it is missing. However, if the others such as WGOSIDS, WGOSISID, WGOSISN are missing, what values or string can be used to replace. This still needs to consider. This needs @BenjaminRuston and @emilyhcliu to give further guidances.

**Comment by YoulongXia-NOAA** [comment]
LSTN always exists without any missing. This is a good alternative.

File: src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (PraveenKumar-NOAA) [reply src/bufr/BufrParser/Exports/Variables/WigosidVariable.cpp:81]: @rmclaren all this difference is coming from ~/bufr/DataObject.h file. DataObject.h is different both in develop and feature/query_new_result. Let us talk about this tomorrow.

**Comment by srherbener** [comment]
@PraveenKumar-NOAA is this PR superseded by #1407? If so, please close this one. Thanks!
